DQM/L1TMonitorClient/src/L1TStage2EMTFDEClient.cc

```diff
@@ -0,0 +1,391 @@
+
+
+
+  for (int bin = 1; bin <= 16; ++bin) {
+    emtfMuonhwQualComp_->setBinLabel(bin, std::to_string(bin - 1), 1);
+  }
+  for (int bin = 1, bin_label = -3; bin <= 7; ++bin, ++bin_label) {
+    emtfMuonhwBXComp_->setBinLabel(bin, std::to_string(bin_label), 1);
+  }
+  //Track Ratio Plots
+  emtfTrackEtaComp_=ibooker.book1D("emtfTrackEtaComp", "2*Data/(Emul+Data) of Track Eta",100,-2.5,2.5);
+  emtfTrackPhiComp_=ibooker.book1D("emtfTrackPhiComp", "2*Data/(Emul+Data) of Track Phi", 128, -3.2, 3.2);
+  emtfTrackPtComp_=ibooker.book1D("emtfTrackPtComp", "2*Data/(Emul+Data) of Track p_{T}", 256, 1, 257);
+  emtfTrackQualComp_=ibooker.book1D("emtfTrackQualComp", "2*Data/(Emul+Data) of Track Quality", 16, 0, 16);
+  emtfTrackModeComp_=ibooker.book1D("emtfTrackModeComp", "2*Data/(Emul+Data) of Track Mode", 16, 0, 16);
+  emtfTrackBXComp_=ibooker.book1D("emtfTrackBXComp", "2*Data/(Emul+Data) of Track BX", 8, -3, 5);
+  
+  emtfTrackEtaCompCoarse_=ibooker.book1D("emtfTrackEtaCompCoarse", "2*Data/(Emul+Data) of Track Eta (coarse)",20,-2.5,2.5);
+  emtfTrackPhiCompCoarse_=ibooker.book1D("emtfTrackPhiCompCoarse", "2*Data/(Emul+Data) of Track Phi (coarse)", 32, -3.2, 3.2);
+  emtfTrackPtCompCoarse_=ibooker.book1D("emtfTrackPtCompCoarse", "2*Data/(Emul+Data) of Track p_{T} (coarse)", 32, 1, 257);
+  
+
+  for (int bin = 1, bin_label = -3; bin <= 8; ++bin, ++bin_label) {
+    emtfTrackBXComp_->setBinLabel(bin, std::to_string(bin_label), 1);
+  }
+  emtfTrackSectorIndexComp_=ibooker.book1D("emtfTrackSectorIndexComp","Data/Emul of Track Sector Index", 13, -6.5, 6.5);
+
+
+  for (int bin = 1; bin <= 16; ++bin) {
+    emtfTrackModeComp_->setBinLabel(bin, std::to_string(bin - 1), 1);
+    emtfTrackQualComp_->setBinLabel(bin, std::to_string(bin - 1), 1);
+  }
+} 
+void L1TStage2EMTFDEClient::fixZero(TH1F* histData, TH1F* histEmul){
+  for (Int_t i=0;i<histData->GetSize();i++){
+    if(histData->GetBinContent(i)==0) histData->SetBinContent(i,0.01);
+  }
+  for (Int_t i=0;i<histEmul->GetSize();i++){
+    if(histEmul->GetBinContent(i)==0) histEmul->SetBinContent(i,0.01);
+  }
+} 
+void L1TStage2EMTFDEClient::processHistograms(DQMStore::IGetter &igetter){
+  
+  MonitorElement* dataHist_;
+  MonitorElement* emulHist_;
+
+  //Ratio Plots are 2*data/(data + emul) so plots go from 0 to 2, where 0=no data, 1=perfect match, 2=no emul
+  //RegionaMuonCand Ratio Plots
+
+  //hwEta
+  dataHist_ = igetter.get(input_dir_data_+"/MuonCand/"+"emtfMuonhwEta");
+  emulHist_ = igetter.get(input_dir_emul_+"/MuonCand/"+"emtfMuonhwEta");
+
+ if (dataHist_ && emulHist_){
+    TH1F *hwEtaD = dataHist_->getTH1F();
+    TH1F *hwEtaE = emulHist_->getTH1F();
+
+    TH1F *hwEtaRatio = emtfMuonhwEtaComp_->getTH1F();
+    TH1F *hwEtaDif = (TH1F*) hwEtaRatio->Clone();
+    L1TStage2EMTFDEClient::fixZero(hwEtaD, hwEtaE); 
+
+    hwEtaDif->Add(hwEtaD, hwEtaE);
+
+    hwEtaRatio->Divide(hwEtaD, hwEtaDif, 2, 1); 
```

Comment:
so this effectively becomes

2*hwEtaD[i] / ( max(0.1+hwEtaD[i]) + max(0.1+hwEtaE) )

Right? Is the double 0.1 needed?